transaction/bind.go

```diff
@@ -43,12 +45,85 @@ func NewHermezAPITxRequest(poolTx *hezcommon.PoolL2Tx, token hezcommon.Token) AP
 	}
 }
 
+func NewSignedAPITxFromEthAddr(chainID int, fromBjjWallet account.BJJWallet, fromIdx int64, toEthAddress string, amount *big.Int, feeSelector hezCommon.FeeSelector, token hezCommon.Token, nonce int) (APITx, error) {
+
+	f40Amount, err := AmountToFloat40(amount)
+	if err != nil {
+		return APITx{}, err
+	}
+
+	tx := &hezCommon.PoolL2Tx{
+		FromIdx:   hezCommon.Idx(fromIdx),
+		ToEthAddr: ethCommon.HexToAddress(toEthAddress),
+		ToBJJ:     hezCommon.EmptyBJJComp,
+		ToIdx:     0,
+		Amount:    f40Amount,
+		Fee:       hezCommon.FeeSelector(uint8(feeSelector)),
+		TokenID:   hezCommon.TokenID(token.TokenID),
+		Nonce:     hezCommon.Nonce(nonce),
+		Type:      hezCommon.TxTypeTransferToEthAddr,
+	}
+
+	return SignAPITx(chainID, fromBjjWallet, token, tx)
+}
+
+func SignAPITx(chainID int, fromBjjWallet account.BJJWallet, token hezCommon.Token, tx *hezCommon.PoolL2Tx) (APITx, error) {
+	tx, err := hezCommon.NewPoolL2Tx(tx)
+	if err != nil {
+		return APITx{}, err
+	}
+
+	txHash, err := tx.HashToSign(uint16(chainID))
+	if err != nil {
+		return APITx{}, err
+	}
+
+	signedTx := fromBjjWallet.PrivateKey.SignPoseidon(txHash)
+	tx.Signature = signedTx.Compress()
+
+	t := hezCommon.Token{
+		TokenID: hezCommon.TokenID(token.TokenID),
+		Symbol:  token.Symbol,
+	}
+
+	apiTxRequest := NewHermezAPITxRequest(tx, t)
+
+	return apiTxRequest, nil
+}
+
 // IdxToHez convert idx to hez idx
-func IdxToHez(idx hezcommon.Idx, tokenSymbol string) string {
+func IdxToHez(idx hezCommon.Idx, tokenSymbol string) string {
 	// log.Printf("idx %+v\ntoken: %s\n", idx, tokenSymbol)
 	return "hez:" + tokenSymbol + ":" + strconv.Itoa(int(idx))
 }
 
+func HezToIdx(hezIdx string) (int64, error) {
```

Comment:
Maybe it exists in hermez node common. In the future it would be nice to use hermez node libraries for those items.